torch/csrc/tensor/python_tensor.cpp

```diff
@@ -434,27 +419,29 @@ static void py_bind_tensor_types(
   }
 }
 
-static bool PyTensorType_Check(PyObject* obj) {
-  auto it = std::find_if(
-      tensor_types.begin(), tensor_types.end(), [obj](PyTensorType* x) {
-        return (PyObject*)x == obj;
-      });
-  return it != tensor_types.end();
-}
-
 void py_set_default_tensor_type(PyObject* obj) {
   TORCH_WARN_ONCE(
       "torch.set_default_tensor_type() is deprecated as of PyTorch 2.1, "
       "please use torch.set_default_dtype() and torch.set_default_device() as alternatives.")
-  TORCH_CHECK_TYPE(
-      PyTensorType_Check(obj),
-      "invalid type object: only floating-point types are supported as the default type");
+  auto get_python_tensor_classes = []() -> THPObjectPtr {
+    auto torch_module = THPObjectPtr(PyImport_ImportModule("torch"));
+    if (!torch_module)
+      throw python_error();
+
+    auto tensor_classes = THPObjectPtr(
+        PyObject_GetAttrString(torch_module.get(), "_tensor_classes"));
+    if (!tensor_classes)
+      throw python_error();
+    return tensor_classes;
+  };
+  // Checked whether passed object is in _tensor_classes set.
+  // Which is defined in torch/__init__.py, and get all supported items by
+  // call torch._tensor_classes in python script.
+  auto torch_tensor_classes = get_python_tensor_classes();
   PyTensorType* type = (PyTensorType*)obj;
   TORCH_CHECK_TYPE(
-      !type->is_cuda || torch::utils::cuda_enabled(),
-      "type ",
-      type->name,
-      " not available. Torch not compiled with CUDA enabled.")
```

Comment:
So what happened to this test? It seems to have vanished?

Reply:
Oh, i overlooked this, add this back.